python/xorbits/_mars/learn/model_selection/_search.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2022-2023 XProbe Inc.
+# derived from copyright 1999-2021 Alibaba Group Holding Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from numbers import Number
+from typing import Dict, Iterable, List, Mapping, Sequence, Union
+
+import numpy as np
+from sklearn.model_selection._search import ParameterGrid as _SK_ParameterGrid
+
+from ... import tensor as mt
+
+
+class ParameterGrid(_SK_ParameterGrid):
+    param_grid: List[Dict[str, Union[mt.Tensor, Sequence]]]
+
+    def __init__(
+        self, param_grid: Union[List[Dict[str, Iterable]], Mapping[str, Iterable]]
+    ):
+        if not isinstance(param_grid, (Mapping, Iterable)):
+            raise TypeError(
+                f"Parameter grid should be a dict or a list, got: {param_grid!r} of"
+                f" type {type(param_grid).__name__}"
+            )
+
+        if isinstance(param_grid, Mapping):
+            param_grid = [param_grid]
+
+        for grid in param_grid:
+            if not isinstance(grid, dict):
+                raise TypeError(f"Parameter grid is not a dict ({grid!r})")
+            for key, value in grid.items():
+                if isinstance(value, (np.ndarray, Number)):
+                    xnp_value = mt.array(value)
+                    value = xnp_value
+                    grid[key] = xnp_value
+                elif isinstance(value, Iterable):
+                    is_num = False
+                    for i, v in enumerate(value):
+                        if isinstance(v, (np.ndarray, Number)):
+                            is_num = True
+                            xnp_value = mt.array(v)
+                            value[i] = xnp_value
+                            grid[key][i] = xnp_value
+                    if is_num:
+                        grid[key] = mt.ExecutableTuple(grid[key])
```

Comment:
Would `grid[key]` contain both strings and numbers?

Reply:
Hi, yes i checked and this can happen. i checked the logic to make ```is_num``` false whenever another type occurs.